### PR TITLE
FIX perform scaling for each runhistory2epm

### PR DESCRIPTION
**Pull request by mfeurer**
So far the scaling was only done for the log transformation. This PR adds scaling to each runhistory and calls a standard function on the runhistory when retrieving the incumbent value.

### CODE REVIEW

File: smac/runhistory/runhistory2epm.py [hunk 1]
```diff
@@ -321,232 +348,127 @@ def _build_matrix(self, run_dict: typing.Mapping[RunKey, RunValue],
...
-        run_dict: dict(RunKey -> RunValue)
-            Dictionary from RunHistory.RunKey to RunHistory.RunValue
-        runhistory: RunHistory
-            Runhistory object
-        instances: list
-            List of instances
-        par_factor: int
-            Penalization factor for censored runtime data
+        #perc = np.percentile(values, self.scale_perc)
+        #min_y = np.min(values)
```
Comment (mlindauer) [inline smac/runhistory/runhistory2epm.py]: remove unused line

File: smac/runhistory/runhistory2epm.py [hunk 2]
```diff
@@ -321,232 +348,127 @@ def _build_matrix(self, run_dict: typing.Mapping[RunKey, RunValue],
...
-            Penalization factor for censored runtime data
-
-        Returns
-        -------
-        X: np.ndarray
-        Y: np.ndarray
-        """
-        X, y = super()._build_matrix(run_dict=run_dict, runhistory=runhistory,
-                                     instances=instances, par_factor=par_factor)
+    def transform_response_values(self, values: np.ndarray) -> np.ndarray:
```
Comment (mlindauer) [inline smac/runhistory/runhistory2epm.py]: please add doc strings

File: smac/runhistory/runhistory2epm.py [hunk 3]
```diff
@@ -321,232 +348,127 @@ def _build_matrix(self, run_dict: typing.Mapping[RunKey, RunValue],
...
-        runhistory: RunHistory
-            Runhistory object
-        instances: list
-            List of instances
-        par_factor: int
-            Penalization factor for censored runtime data
+        #perc = np.percentile(values, self.scale_perc)
+        #min_y = np.min(values)
+        min_y = self.min_y - (self.perc - self.min_y)  # Subtract the difference between the percentile and the minimum
+        #max_y = np.max(values)
```
Comment (mlindauer) [inline smac/runhistory/runhistory2epm.py]: same here

File: smac/runhistory/runhistory2epm.py [hunk 4]
```diff
@@ -180,7 +187,7 @@ def transform(self, runhistory: RunHistory):
         # Store a list of instance IDs
         s_instance_id_list = [k.instance_id for k in s_run_dict.keys()]
         X, Y = self._build_matrix(run_dict=s_run_dict, runhistory=runhistory,
-                                  instances=s_instance_id_list)
+                                  instances=s_instance_id_list, store_statistics=True)
```
Comment (mlindauer) [inline smac/runhistory/runhistory2epm.py]: if our first run is a timeout, we will never compute the statistics

**Comment by mfeurer** [comment]
ping @mlindauer

File: smac/optimizer/smbo.py [hunk 5]
```diff
@@ -304,8 +304,9 @@ def _get_incumbent_value(self):
                 raise ValueError("Runhistory is empty and the cost value of "
                                  "the incumbent is unknown.")
             incumbent_value = self.runhistory.get_cost(self.incumbent)
-            if isinstance(self.rh2EPM,RunHistory2EPM4LogCost):
-                incumbent_value = np.log(incumbent_value)
+            # It's unclear how to do this for inv scaling and potential future scaling. This line should be changed if
+            # necessary
+            incumbent_value = self.rh2EPM.transform_response_values(np.array(incumbent_value).reshape((1, 1)))[0][0]
```
Comment (KEggensperger) [inline smac/optimizer/smbo.py]: Does this really have to look that complicated?

File: smac/runhistory/runhistory2epm.py [hunk 6]
```diff
@@ -130,11 +129,17 @@ def __init__(
                              "smac.epm.base_imputor.BaseImputor, but %s" %
                              type(self.imputor))
 
+        # Learnt statistics
```
Comment (KEggensperger) [inline smac/runhistory/runhistory2epm.py]: It's "learned"

**Pull request merged**